### PR TITLE
Commit for Issue#5

### PR DESCRIPTION
**Pull request by saisumanv**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -336,14 +342,55 @@ def issueToken(payload, algorithm="RS256"):
         token['aud'] = "https://demo.scitokens.org"
 
     serialized_token = token.serialize(issuer = "https://demo.scitokens.org", lifetime = lifetime)
-    return serialized_token
+    print(serialized_token)
```
Comment (djw8605) [inline app.py]: I think this is a debugging line, go ahead and remove in production.

File: js/app.js [hunk 2]
```diff
@@ -548,32 +552,73 @@ FaFp+DyAe+b4nDwuJaW2LURbr8AEZga7oQj0uYxcYw==\n\
...
+    }*/
+    
 
     var result = false;
     if (algorithm == "RS256") {
+		//Get the pasted token
+		newToken = tokenEditor.getValue();
+		//newToken = "blah";
+		
+      $.ajax({
```
Comment (djw8605) [inline js/app.js:568]: Really odd spacing.  Are these tabs?  I think JS is usually 2 or 4 spaces.  The rest of the file tabbing is 2 spaces.  Please update!

File: app.py [hunk 3]
```diff
@@ -20,9 +20,15 @@
...
-r = redis.from_url(os.environ.get("REDIS_URL"))
 
+import jwt#sai
+from jwt import PyJWKClient#sai- need to run "pip3 install -U pyjwt" because jwt and pyjwt modules conflict
+
+
+logging.basicConfig(level=logging.DEBUG)
+#print(os.environ.get("REDIS_URL"))
+#r = redis.from_url(os.environ.get("REDIS_URL")) or "redis://"
+r = "redis://"
```
Comment (djw8605) [inline app.py]: This needs to go back to the regular redis command.

File: app.py [hunk 4]
```diff
@@ -20,9 +20,15 @@
...
 from pottery import synchronize
-logging.basicConfig(level=logging.DEBUG)
-r = redis.from_url(os.environ.get("REDIS_URL"))
 
+import jwt#sai
+from jwt import PyJWKClient#sai- need to run "pip3 install -U pyjwt" because jwt and pyjwt modules conflict
+
+
+logging.basicConfig(level=logging.DEBUG)
+#print(os.environ.get("REDIS_URL"))
```
Comment (djw8605) [inline app.py]: Remove this debugging print line.

File: app.py [hunk 5]
```diff
@@ -290,7 +296,7 @@ def Issue():
         except json.decoder.JSONDecodeError as json_err:
             return "", 400
     
-    return issueToken(payload, algorithm)
+    return ""##issueToken(payload, algorithm)##sai uncomment
```
Comment (djw8605) [inline app.py]: Why is this return commented out?

File: app.py [hunk 6]
```diff
@@ -336,14 +342,55 @@ def issueToken(payload, algorithm="RS256"):
...
-
+@app.route('/verify',methods=['POST'])
+def Verify():
+    """
+    Verify the SciToken received from the client
+    """
+    if request.method == 'POST':
+        data=request.data
+        dataDict = json.loads(data)
+        print(dataDict['token'])
```
Comment (djw8605) [inline app.py]: Probably shouldn't print the token.

File: js/app.js [hunk 7]
```diff
@@ -390,7 +390,7 @@ FaFp+DyAe+b4nDwuJaW2LURbr8AEZga7oQj0uYxcYw==\n\
   })
 
   function updateCurlCommand(serialized_token) {
-    curlCommand.setValue("curl -H \"Authorization: Bearer " + serialized_token + "\" https://demo.scitokens.org/protected")
+    curlCommand.setValue("curl -H \"Authorization: Bearer " + serialized_token + "\" https://demo.scitokens.org/protected")//sai
```
Comment (djw8605) [inline js/app.js]: What changed here?

**Review by djw8605 — CHANGES_REQUESTED** [review]
Bunch of changes.

File: js/app.js (hunk 2 above)
Comment (saisumanv) [reply js/app.js:568]: Updated.

File: app.py (hunk 3 above)
Comment (saisumanv) [reply app.py]: Ok. Not sure if this matters but I had issues using the original REDIS URL on my Ubuntu (via WSL2). A little digging told me that it has to do with my Ubuntu+REDIS version.

File: js/app.js (hunk 2 above)
Comment (djw8605) [reply js/app.js:568]: Still has odd tabbing.

**Pull request merged**